drivers/hmis/app/models/hmis/project_group_criteria.rb

```diff
@@ -0,0 +1,168 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+module Hmis
+  class ProjectGroupCriteria < ::ModelForm
+    include ApplicationHelper
+    include ActionView::Helpers::TagHelper
+    include ActionView::Context
+    include ::Hmis::Concerns::HmisArelHelper
+
+    # Source data source for this project filter. ProjectGroupCriteria filter cannot be used across multiple data sources.
+    attribute :data_source_id, Integer, default: nil
+
+    # Attributes for filtering
+    attribute :project_ids, Array, default: []
+    attribute :organization_ids, Array, default: []
+    attribute :all_projects_in_data_source, Boolean, default: false
```

Comment:
Maybe a minor re-factor for a future date, but it seems like choosing `all_projects_in_data_source` should be mutually exclusive with the other options.